package/boot/arm-trusted-firmware-mvebu/Makefile

```diff
@@ -0,0 +1,112 @@
+#
+# Copyright (C) 2019 OpenWrt
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=arm-trusted-firmware-mvebu
+PKG_RELEASE:=2.1
+
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_RELEASE).tar.xz
+PKG_SOURCE_URL=https://github.com/ARM-software/arm-trusted-firmware
+PKG_SOURCE_DATE:=2019-10-21
+PKG_SOURCE_VERSION:=937f669872b5516c9a0fd3d60f9618ad54a72b5c
+PKG_MIRROR_HASH:=ca7e6b148f16064c99303b7e0f9197ecf20b61793b0a947fc454852629dabda1
+
+PKG_MAINTAINER:=Vladimir Vid <vladimir.vid@sartura.hr>
+
+include $(INCLUDE_DIR)/package.mk
+
+define Package/arm-trusted-firmware-mvebu
+    SECTION:=boot
+    CATEGORY:=Boot Loaders
+    TITLE:=ARM Trusted Firmware for mvebu devices
+    DEPENDS:=@TARGET_mvebu
+endef
+
+define Package/arm-trusted-firmware-mvebu/uDPU
+$(call Package/arm-trusted-firmware-mvebu)
+    BOARDNAME:=uDPU
+    TITLE:=ARM Trusted Firmware for Methode uDPU
+    DEPENDS:=+u-boot-uDPU
+    UBOOT:=uDPU
+    CLOCKSPRESET:=CPU_1000_DDR_800
+    PLAT:=a3700
+endef
+
+A3700_UTILS_NAME:=a3700-utils
+A3700_UTILS_RELEASE:=18.12.1
+A3700_UTILS_SOURCE=$(A3700_UTILS_NAME)-$(A3700_UTILS_RELEASE).tar.bz2
+
+define Download/a3700-utils
+  FILE:=$(A3700_UTILS_SOURCE)
+  PROTO:=git
+  URL:=https://github.com/MarvellEmbeddedProcessors/A3700-utils-marvell.git
+  VERSION:=e6bb176fb9936d9160e53ae09275000dc6440686
+  MIRROR_HASH:=c3edf71cbf1fc2e5482d36e36a34c79d682d2c87fdfb909b4dc6327a171e7bd4
+  SUBDIR:=$(A3700_UTILS_NAME)
+endef
+
+MV_DDR_NAME:=mv-ddr-marvell
+MV_DDR_RELEASE:=mainline
+MV_DDR_SOURCE:=$(MV_DDR_NAME)-$(MV_DDR_RELEASE).tar.bz2
+
+define Download/mv-ddr-marvell
+  FILE:=$(MV_DDR_SOURCE)
+  PROTO:=git
+  URL:=https://github.com/MarvellEmbeddedProcessors/mv-ddr-marvell.git
+  VERSION:=a881467ef0f0185e6570dd0483023fde93cbb5f5
+  MIRROR_HASH:=19f34817616c583b2b5f2612705857cf2120f6c2132355217899b1403c3fd336
+  SUBDIR:=$(MV_DDR_NAME)
+endef
+
+LINARO_NAME:=gcc-linaro
+LINARO_RELEASE:=6
+LINARO_VERSION:=5.0-2018.12-$(HOST_ARCH)_arm-linux-gnueabi
+LINARO_SOURCE=$(LINARO_NAME)-$(LINARO_RELEASE).$(LINARO_VERSION).tar.xz
+
+define Download/gcc-linaro
+  FILE:=$(LINARO_SOURCE)
+  URL:=https://releases.linaro.org/components/toolchain/binaries/latest-$(LINARO_RELEASE)/arm-linux-gnueabi/
```

Comment:
> 32-bit cross compiler is used for building the WTMI image.

What about providing binary WTMI images instead? I don't like this idea.

Reply:
Not sure this would be the best solution as we cannot use a single binary image for all devices and we would eventually end up bloated with binaries.
From my point of view, not a lot of people will want to update their bootloader and there aren't any dependencies on the target devices for this package, so is there any particular reason you're against this?

Reply:
>  we would eventually end up bloated with binaries.

So now we ended up bloated with Linux only compilers AND binaries.

> so is there any particular reason you're against this?

1. Unable to build this package on macOS host (yeah, we still support this use case)
2. Nothing depends on this package in the tree, so it probably belongs to the package feeds
3. I don't like this idea of external toolchain download in general, because it screams "support nigthmare"